app/workers/functions.php

```diff
@@ -463,25 +436,36 @@ public function execute(string $trigger, string $projectId, string $executionId,
         $stderr = '';
 
         $executionStart = \microtime(true);
-        
-        $exitCode = Console::execute("docker exec ".\implode(" ", $vars)." {$container} {$command}"
-            , '', $stdout, $stderr, $function->getAttribute('timeout', (int) App::getEnv('_APP_FUNCTIONS_TIMEOUT', 900)));
+
+        $exitCode = 0;
+
+        try {
+            $exitCode = $orchestration->execute(
+                name: $container, 
+                command: $orchestration->parseCommandString($command), 
+                stdout: $stdout, 
+                stderr: $stderr, 
+                vars: $vars, 
+                timeout: $function->getAttribute('timeout', (int) App::getEnv('_APP_FUNCTIONS_TIMEOUT', 900)));
+        } catch (TimeoutException $e) {
+            $exitCode = 0;
+        }
 
         $executionEnd = \microtime(true);
         $executionTime = ($executionEnd - $executionStart);
-        $functionStatus = ($exitCode === 0) ? 'completed' : 'failed';
+        $functionStatus = $exitCode ? 'completed' : 'failed';
```

Comment:
Exit code 0 indicates a **successful** operation in UNIX, and failures are `0 < $exitCode < 256`. but this logic uses `$exitCode = 0` to indicate failure.

Let's update the logic: `$exitCode = 0` for successful operations, `$exitCode = 124` for timeouts